### PR TITLE
Minimal Footer Updates

### PR DESCRIPTION
**Pull request by JoshBowdenConcepts**
## Summary

There were some layout issues adding polymorphism to the MinimalFooter.Link component. To address this I have updated the CSS to reflect the changes and updated the Storybook docs so we will have visual regression tests.

## List of notable changes:

- **updated** MinimalFooter.Link CSS to specify padding on all sides to avoid browser-implied styles for buttons.

## What should reviewers focus on?

- Ensuring no additional padding is added for the MinimalFooter.Link component when using it as a Button

## Steps to test:

1. Open the MinimalFooter component in CI-deployed preview environment
2. Go to Mixed Buttons and LInks story in Storybook
3. Verify that each link item behaves as expected.

## Supporting resources (related issues, external links, etc):

- [Slack Thread](https://github.slack.com/archives/C03FY9G00NS/p1684171676600209)

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

File: packages/react/src/MinimalFooter/MinimalFooter.features.stories.tsx [hunk 1]
```diff
@@ -61,9 +61,10 @@ export const MaximumLinks = () => (
 )
 MaximumLinks.storyName = 'Maximum Links (5)'
 
-export const ButtonLinks = () => (
+export const MixedButtonsAndLinks = () => (
   <MinimalFooter socialLinks={false}>
-    <MinimalFooter.Link as="button" onClick={() => alert('You have clicked Button one')}>
+    <MinimalFooter.Link>Link one</MinimalFooter.Link>
+    <MinimalFooter.Link as="button" href="true" onClick={() => alert('You have clicked Button one')}>
```
Comment (rezrah) [inline packages/react/src/MinimalFooter/MinimalFooter.features.stories.tsx:67]: What is the a href="true" for? It looks like the link above it is also missing a href="#" to make this a mixed use story?

Comment (JoshBowdenConcepts) [reply packages/react/src/MinimalFooter/MinimalFooter.features.stories.tsx]: This has been updated in this [SHA](https://github.com/primer/brand/pull/280/commits/9ce80ac7b2f3c052c64d3d432177522034703537)

File: packages/react/src/MinimalFooter/MinimalFooter.features.stories.tsx [hunk 2]
```diff
@@ -61,9 +61,10 @@ export const MaximumLinks = () => (
 )
 MaximumLinks.storyName = 'Maximum Links (5)'
 
-export const ButtonLinks = () => (
+export const MixedButtonsAndLinks = () => (
   <MinimalFooter socialLinks={false}>
-    <MinimalFooter.Link as="button" onClick={() => alert('You have clicked Button one')}>
+    <MinimalFooter.Link>Link one</MinimalFooter.Link>
+    <MinimalFooter.Link as="a" href="#" onClick={() => alert('You have clicked Button one')}>
```
Comment (rezrah) [inline packages/react/src/MinimalFooter/MinimalFooter.features.stories.tsx:67]: @JoshBowdenConcepts I'm still seeing `href` and `onClick` (in the context of a button) being applied at the same time? Is this supposed to be an `a` or a `button`? Given the label is `Button one`, recommend you remove the `as="a" href="#"`.

```suggestion
    <MinimalFooter.Link as="button" onClick={() => alert('You have clicked Button one')}>
```

File: packages/react/src/MinimalFooter/MinimalFooter.features.stories.tsx [hunk 3]
```diff
@@ -61,9 +61,10 @@ export const MaximumLinks = () => (
 )
 MaximumLinks.storyName = 'Maximum Links (5)'
 
-export const ButtonLinks = () => (
+export const MixedButtonsAndLinks = () => (
   <MinimalFooter socialLinks={false}>
-    <MinimalFooter.Link as="button" onClick={() => alert('You have clicked Button one')}>
+    <MinimalFooter.Link>Link one</MinimalFooter.Link>
```
Comment (rezrah) [inline packages/react/src/MinimalFooter/MinimalFooter.features.stories.tsx:66]: As the default tag is `a`, this should have require a `href`.

File: packages/react/src/MinimalFooter/MinimalFooter.features.stories.tsx (hunk 1 above)
Comment (rezrah) [reply packages/react/src/MinimalFooter/MinimalFooter.features.stories.tsx]: Left a [follow up comment](https://github.com/primer/brand/pull/280#discussion_r1202097001) here.

**Review by rezrah — COMMENTED** [review]
Hey @JoshBowdenConcepts. This is looking good. Could you please add a changeset to this PR indicating what's being fixed.

**Review by rezrah — APPROVED** [review]
👍 Looks good.

One final change before we merge this please:

Screenshot 2023-05-25 at 12 50 18

The default is `a`, so could you update the docs to indicate this please? Shouldn't be `undefined`.

**Pull request merged**